src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,101 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private Node<K, V>[] array;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        array = new Node[DEFAULT_CAPACITY];
+        threshold = (int) (DEFAULT_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (++size > threshold) {
+            array = resize();
+        }
+        addElement(key, value, array);
     }
 
     @Override
     public V getValue(K key) {
+        int index = Math.abs(key == null ? 0 : (key.hashCode() % array.length));
+        Node<K, V> current = array[index];
+        while (current != null) {
+            if (Objects.equals(current.key, key)) {
+                return current.value;
+            }
+            current = current.next;
+        }
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private void addElement(K key, V value, Node<K, V>[] currentArray) {
+        int keyHashCode = key == null ? 0 : key.hashCode();
+        int index = keyHashCode % currentArray.length;
```

Comment:
Repeatable calculations (line 27 as well)

Reply:
It isn't same because I in this method I find keyHashCode, I use it in line 65, so as not to search for the hash code again.